apiserver/facades/client/client/status_test.go

```diff
@@ -377,6 +377,62 @@ func (s *statusUnitTestSuite) TestMigrationInProgress(c *gc.C) {
 	setAndCheckMigStatus("oh noes")
 }
 
+func (s *statusUnitTestSuite) TestRelationFiltered(c *gc.C) {
+	// make application 1 with endpoint 1
+	a1 := s.Factory.MakeApplication(c, &factory.ApplicationParams{
+		Name: "abc",
+		Charm: s.Factory.MakeCharm(c, &factory.CharmParams{
+			Name: "wordpress",
+		}),
+	})
+	e1, err := a1.Endpoint("db")
+	c.Assert(err, jc.ErrorIsNil)
+
+	// make application 2 with endpoint 2
+	a2 := s.Factory.MakeApplication(c, &factory.ApplicationParams{
+		Name: "def",
+		Charm: s.Factory.MakeCharm(c, &factory.CharmParams{
+			Name: "mysql",
+		}),
+	})
+	e2, err := a2.Endpoint("server")
+	c.Assert(err, jc.ErrorIsNil)
+
+	// create relation between a1 and a2
+	r12 := s.Factory.MakeRelation(c, &factory.RelationParams{
+		Endpoints: []state.Endpoint{e1, e2},
+	})
+	c.Assert(r12, gc.NotNil)
+
+	// create another application 3 with an endpoint 3
+	a3 := s.Factory.MakeApplication(c, &factory.ApplicationParams{
+		Charm: s.Factory.MakeCharm(c, &factory.CharmParams{Name: "logging"}),
+	})
+	e3, err := a3.Endpoint("info")
+	c.Assert(err, jc.ErrorIsNil)
+
+	// create endpoint 4 on application 1
+	e4, err := a1.Endpoint("juju-info")
+	c.Assert(err, jc.ErrorIsNil)
+	r13 := s.Factory.MakeRelation(c, &factory.RelationParams{
+		Endpoints: []state.Endpoint{e3, e4},
+	})
+	c.Assert(r13, gc.NotNil)
+
+	// Test status filtering with application 1: should get both relations
+	client := s.APIState.Client()
+	status, err := client.Status([]string{a1.Name()})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(status, gc.NotNil)
+	c.Assert(status.Relations, gc.HasLen, 2)
```

Comment:
Shouldn't we be asserting *what* relations are in there, and not just the length?